news/management/commands/process_fxa_data.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import print_function, unicode_literals
+
+from email.utils import formatdate
+
+from django.conf import settings
+from django.core.cache import cache
+from django.core.management import BaseCommand, CommandError
+
+import boto3
+from apscheduler.schedulers.blocking import BlockingScheduler
+from django_statsd.clients import statsd
+from pathlib2 import Path
+from pytz import utc
+from raven.contrib.django.raven_compat.models import client as sentry_client
+
+from news.backends.sfmc import sfmc, NewsletterException
+
+
+TMP = Path('/tmp')
+BUCKET_DIR = 'fxa-last-active-timestamp/data'
+DATA_PATH = TMP.joinpath(BUCKET_DIR)
+FXA_IDS = {}
+FILE_DONE_KEY = 'fxa_activity:completed:%s'
+TWO_WEEKS = 60 * 60 * 24 * 14
+schedule = BlockingScheduler(timezone=utc)
+
+
+def _fxa_id_key(fxa_id):
+    return 'fxa_activity:%s' % fxa_id
+
+
+def get_fxa_time(fxa_id):
+    fxatime = FXA_IDS.get(fxa_id)
+    if fxatime is None:
+        fxatime = cache.get(_fxa_id_key(fxa_id))
+        if fxatime:
+            FXA_IDS[fxa_id] = fxatime
+
+    return fxatime or 0
+
+
+def set_fxa_time(fxa_id, fxa_time):
+    try:
+        sfmc.upsert_row(settings.FXA_SFMC_DE, {
+            'FXA_ID': fxa_id,
+            'Timestamp': formatdate(timeval=fxa_time, usegmt=True),
+        })
+    except NewsletterException:
+        sentry_client.captureException()
+        # try again later
```

Comment:
Later would be the next time this `fxa_id` appears in a new logfile in s3? 

Could we make this a celery async task and automatically retry a couple of times or is this over-engineered for the type of data we have?